### PR TITLE
E2E tests for horizontal pod autoscaler.

### PR DESCRIPTION
**Pull request by jszczepkowski**
Fixes & tuning in horiontal pod autoscaler and its e2e tests; two of the tests added to "Autoscaling suite".

### CODE REVIEW

**Comment by jszczepkowski** [comment]
Part of #14108 & #12087.

File: test/e2e/autoscaling_utils.go [hunk 1]
```diff
@@ -227,6 +227,8 @@ func (rc *ResourceConsumer) WaitForReplicas(desiredReplicas int) {
 func (rc *ResourceConsumer) CleanUp() {
 	rc.stopCPU <- 0
 	rc.stopMem <- 0
+	// Wait some time to ensure all child goroutines are finished.
+	time.Sleep(2 * rc.sleepTime)
```
Comment (piosz) [inline test/e2e/autoscaling_utils.go]: Change here to 10s as discussed offline

**Comment by piosz** [comment]
LGTM

File: test/e2e/autoscaling_utils.go (hunk 1 above)
Comment (jszczepkowski) [inline test/e2e/autoscaling_utils.go]: Done.

**Comment by pmorie** [comment]
@kubernetes/rh-scalability

**Pull request merged**